sdk/aqueduct/decorator.py

```diff
@@ -127,6 +137,40 @@ def wrap_spec(
         ],
     )
 
+    if associate_system_metrics:
+        assert type(output_artifact) is TableArtifact
+        for system_metric_name, description in SYSTEM_METRICS_INFO:
+            system_operator_id = generate_uuid()
+            system_output_artifact_id = generate_uuid()
+            metric_name = "%s %s metric" % (op_name, system_metric_name)
+            metric_spec = SystemMetricSpec(metricname=system_metric_name)
+            metric_artifact_spec = ArtifactSpec(float={})
+
+            apply_deltas_to_dag(
+                dag,
+                deltas=[
+                    AddOrReplaceOperatorDelta(
+                        op=Operator(
+                            id=system_operator_id,
+                            name=metric_name,
+                            description=description,
+                            spec=OperatorSpec(systemmetric=metric_spec),
+                            inputs=[output_artifact_id],
+                            outputs=[system_output_artifact_id],
+                        ),
+                        output_artifacts=[
+                            Artifact(
+                                id=system_output_artifact_id,
+                                name=artifact_name_from_op_name(metric_name),
+                                spec=metric_artifact_spec,
+                            )
+                        ],
+                    ),
+                ],
+            )
```

Comment:
hmm why we do add every system metric to the dag here, instead of when the user requests it in `.system_metric()`. It seems somewhat unnecessary to always add it to the dag and force a calculation.

Reply:
yeah, i moved it. I forgot we could request the calling op from the dag itself.